src/main/scala/inox/solvers/SolverFactory.scala

```diff
@@ -213,25 +215,26 @@ object SolverFactory {
           protected val underlying = new {
             val program: progEnc.targetProgram.type = progEnc.targetProgram
             val context = ctx
-          } with smtlib.Z3Solver {
+          } with smtlib.optimization.Z3Optimizer {
             val semantics: program.Semantics = targetSem
           }
         }
       })
 
-      case "smt-z3-opt" => create(p)(finalName, {
+      case _ if finalName.startsWith("smt-z3") => create(p)(finalName, {
```

Comment:
I don't really like the fact that the case ordering becomes important. 

Could you maybe try with a more precise pattern?
```scala
case "smt-z3" | r"smt-z3:.+" => ...
```

Reply:
`r""` doesn't seem to work (does it require an import?), I ended up using:
```scala
case _ if finalName == "smt-z3" || finalName.startsWith("smt-z3:")
```

Reply:
That's also fine.